### PR TITLE
Add an input validation

### PR DESCRIPTION
**Pull request by HWaruguru**
I have added a validation method that;
- validates using provided field pattern
- validate when the pattern is null and the missing field name includes "url" I have added logic shared in the [Phabricator  Task](https://phabricator.wikimedia.org/T330981)

What I am struggling with is:
- How to clear validations when the user starts typing on input but then clicks the 'Skip to Next' button
- How to disable the 'submit' button when input is invalid
I would appreciate ideas

### CODE REVIEW

File: src/components/UserContributionForm.vue [hunk 1]
```diff
@@ -24,6 +27,15 @@ const missingFieldValue = ref(null);
...
+    alert(
+      "You must be logged in to submit a task. Please click on the Login button."
+    );
+    return;
+  }
+
+  // validate missingFieldValue
+
   const fieldValue = missingFieldValue.value;
   const contributionRecord = {
```
[the remark this answers is not in the document]
Comment (nbarnabee) [reply src/components/UserContributionForm.vue:40]: It's a simple solution, to the "disable the submit button" problem, anyway.  Not the most elegant.

[the remark this answers is not in the document]
Comment (nbarnabee) [reply src/components/UserContributionForm.vue:40]: Another option would be to put `v-bind:disabled="missingFieldValue ? false : true"` on the submit v-btn at line 150

**Comment by nbarnabee** [comment]
The validation rules look good but they don't actually seem to prevent form submission... god I hate Vuetify.

I'm also not having a problem with clearing the validation when I hit "skip to next."

**Comment by nbarnabee** [comment]
Ok, solved the problems with the input validation reset by adding at line 27: `const form = ref(null);`  and at 77 (within the `skipToNext()` function: `form.value.reset();`

**Comment by nbarnabee** [comment]
Opened a PR to be merged onto this branch that adds my fixes:  #43

**Comment by HWaruguru** [comment]
Oh awesome the @nbarnabee PR solves this. Thank you.

**Review by Damilare1 — APPROVED** [review]

**Pull request merged**